substrate-node/pallets/pallet-smart-contract/spec.md

```diff
@@ -8,64 +8,45 @@ Two main components will play a role in achieving a decentralised consensus betw
 
 2: TFGrid Smart Contract
 
-The TFGrid Substrate Database will keep a record of all users, twins, nodes and farmers in the TF Grid network. This makes it easy to integrate the Smart Contract on Substrate as well since we can read from that storage in Runtime.
+The TFGrid Substrate Database will keep a record of all users, twins, nodes and farmers in the TF Grid network. It will also keep a record of all the contracts that are created between users and farmers. This pallet will also be responsible for the billing of these contracts.
 
 check flow diagram: [flow](./flow.png)
 
 The Smart Contract on Substrate will work as following:
 
-## 1: The user wants to deploy a workload, he interacts with this smart contract pallet and calls: `create_contract` with the input being:
+### 1: The user wants to deploy a workload, he interacts with this smart contract pallet and calls: `create_contract` with the input being:
 
-The user must instruct his twin to create the contract. A contract will always belong to a twin and to a node. This relationship is important because only the user's twin can update the contract as well as only the target node can update as well.
+The user must instruct the chain to create the contract. A contract will always belong to a twin and to a node. This relationship is important because only the user's twin can update the contract as well as only the target node can update as well.
```

Comment:
I don't understand the last part? Isn't it only the user's twin that can update the contract? The node itself is not allowed to update the contract, right?

Reply:
The reason I changed this from "his twin" to "the chain" is that it previously sounded like a twin was some kind of program that could be instructed to do something. So I changed to it clarify that the user should undertake such action himself.

Reply:
No I meant the last part. After 'This relationship', the sentence doesn't seem correct, no?

Reply:
I rephrased it, can you check? :)